x-pack/plugin/core/src/test/java/org/elasticsearch/xpack/core/indexlifecycle/ShrinkSetAliasStepTests.java

```diff
@@ -71,15 +72,33 @@ public ShrinkSetAliasStep copyInstance(ShrinkSetAliasStep instance) {
     }
 
     public void testPerformAction() {
-        IndexMetaData indexMetaData = IndexMetaData.builder(randomAlphaOfLength(10)).settings(settings(Version.CURRENT))
-            .numberOfShards(randomIntBetween(1, 5)).numberOfReplicas(randomIntBetween(0, 5)).build();
+        IndexMetaData.Builder indexMetaDataBuilder = IndexMetaData.builder(randomAlphaOfLength(10)).settings(settings(Version.CURRENT))
+            .numberOfShards(randomIntBetween(1, 5)).numberOfReplicas(randomIntBetween(0, 5));
+        AliasMetaData.Builder aliasBuilder = AliasMetaData.builder(randomAlphaOfLengthBetween(3, 10));
+        if (randomBoolean()) {
+            aliasBuilder.routing(randomAlphaOfLengthBetween(3, 10));
+        }
+        if (randomBoolean()) {
+            aliasBuilder.searchRouting(randomAlphaOfLengthBetween(3, 10));
+        }
+        if (randomBoolean()) {
+            aliasBuilder.indexRouting(randomAlphaOfLengthBetween(3, 10));
+        }
```

Comment:
Can we randomly set `is_write_index` as well to make sure that is not copied?

Reply:
it is set on line 89.

Reply:
could be updated to also be null, but I think it is fine if it is always set to a boolean value for the tests

Reply:
Sorry I had missed that